inlang/source-code/sdk2/src/project/loadProjectFromDirectory.ts

```diff
@@ -18,15 +21,56 @@ export async function loadProjectFromDirectory(
 		"blob"
 	>
 ) {
-	const project = await loadProjectInMemory({
+	const tempProject = await loadProjectInMemory({
 		// pass common arguments to loadProjectInMemory
 		...args,
 		blob: await newProject(),
 	});
-	await copyFiles({ fs: args.fs, path: args.path, lix: project.lix });
+	await copyFiles({ fs: args.fs, path: args.path, lix: tempProject.lix });
+
+	// TODO loadMessages - since settings are not reactiv yet - just reload the project
+	const project = await loadProjectInMemory({
+		// pass common arguments to loadProjectInMemory
+		...args,
+		blob: await tempProject.toBlob(),
+	});
+
+	for (const plugin of project.plugins.get()) {
+		// TODO loadMessages - make sure that we have configured either loadMessages and saveMessages or import export
+		// TODO loadMessages - make sure only one pair is defined?
+		if (plugin.loadMessages !== undefined) {
+			// TODO loadMessages - inserting messages will generate a change for every record
+			await loadLegacyMessages({
+				project,
+				fs: args.fs,
+				loadMessagesFn: plugin.loadMessages,
+			});
+		}
+	}
+
 	return project;
 }
 
+async function loadLegacyMessages(args: {
+	project: Awaited<ReturnType<typeof loadProjectInMemory>>;
+	loadMessagesFn: Required<InlangPlugin>["loadMessages"]; 
```

Comment:
Try this

```suggestion
	loadMessagesFn: Required<InlangPlugin>["loadMessages"]; 
  loadMessagesFn: NonNullable<InlangPlugin["loadMessages"]>;
```

Reply:
Required actually worked - missed to drop the todo